src/libraries/System.Private.Xml/src/System/Xml/Core/XmlEncodedRawTextWriterAsync.cs

```diff
@@ -73,6 +73,62 @@ internal override Task WriteXmlDeclarationAsync(string xmldecl)
             return Task.CompletedTask;
         }
 
+        protected override async ValueTask DisposeAsyncCore()
+        {
+            try
+            {
+                await FlushBufferAsync().ConfigureAwait(false);
+            }
+            finally
+            {
+                // Future calls to Close or Flush shouldn't write to Stream or Writer
+                writeToNull = true;
+
+                if (stream != null)
```

Comment:
can this whole if statement be moved outside of the outer finally?

Reply:
> can this whole if statement be moved outside of the outer finally?

No, then resources wouldn't be closed in case exception is thrown from `FlushBufferAsync()` call above

> What happens if there is an exception there?

Not sure about where you are asking, it seems all layers of this nested try-finally are well handled the exception case but using finally